### DISCUSSION TITLE
Creating an input field that responds to Sums?

### DISCUSSION THREAD

**Discussion by Aetheraeons** [mProjectsCode/obsidian-meta-bind-plugin#488] [Q&A]
Hello! First time trying something out like this, very inexperienced :(. 
First, context (unfortunately neccessary): The goal is to create a field / set of fields to represent an TTRPG character's Current and Maximum Health. I want there to be a dynamic way to change it in _frontmatter_. 

- So far, I have a table that configures the **maximum health** (image 1).
Max-Mod adds a bonus / detriment to the maximum health; Temp HP functions separately.
![image](https://github.com/user-attachments/assets/dd3a1761-e139-40a7-87a7-4891d6d6d992)
`VIEW[{HP.Roll} + {Mod.Con} * {Level} + {HP.MaxMod}][math:HP.Max]`

- The strategy at the moment to display Current Health looks like this: 
![image](https://github.com/user-attachments/assets/3c0c8348-8604-4741-b265-29298433fc3a)
`VIEW[{HP.Max} - {HP.Damaged}][math]`

My desire is to create something similar (though nowhere near as complex) as D&DBeyond's Hit Point Tracker: 
![image](https://github.com/user-attachments/assets/efb39c47-8e97-44f3-a86e-bdc11c661fe4)

Effectively, I'm looking for a way to:
- Keep the Max HP field a __formula__
- Have Current HP as a field drawing from it as the base 
- Have some way (potentially separate) to input a number (instance of damage/healing) and have that change the Current HP, with those numbers being able to be "stacked" (essentially we're looking at something being able to process a sum).

So far, I've been alternating between the manual method (fine with smaller numbers but requires counting and continuously remembering to add it all together), and a progress bar i made: 
![image](https://github.com/user-attachments/assets/6546297c-bdab-4690-9c4a-676b5b64a7b9)
`INPUT[progressBar(minValue(0), maxValue(42)):HP.Max]`
Arguably the Progress bar works rather well, even though I do need to manually move it in response to Damage or Healing. Unfortunately, I would have to manually adjust it _whenever the MaxHP changes_, as that input doesn't allow anything but Numbers.

Any advice would be helped! If anyone's seen similar things made, I'd love a link to them.

**Comment by paulharman** [comment]
Did you manage to get anywhere with this?
